File: bench_pg/NET/Core6/Models/Retailer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace RetailerApi.Models;

[Table("retailer")]
public class Retailer
{
  [Column("id")]
  [JsonPropertyName("id")]
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public int Id { get; set; } = -1;
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public string GSTIN { get; set; } = "";
  [Column("Business_name")]
  [JsonPropertyName("business_name")]
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public string BusinessName { get; set; } = "";
  [Column("Contact_person")]
  [JsonPropertyName("contact_person")]
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public string ContactPerson { get; set; } = "";
  [Column("Contact_number")]
  [JsonPropertyName("contact_number")]
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public int ContactNumber { get; set; } = -1;
  [Column("Contact_address")]
  [JsonPropertyName("contact_address")]
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public string ContactAddress { get; set; } = "";
  [Column("Contact_emailId")]
  [JsonPropertyName("contact_emailId")]
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public string ContactEmailId { get; set; } = "";
  // [Column("Status")]
  [JsonPropertyName("status")]
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public string Status { get; set; } = "";
  [Column("Outlet_limit")]
  [JsonPropertyName("outlet_limit")]
  [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
  public int OutletLimit { get; set; } = -1;
}

File: bench_maria/Elixir/retailer_api/lib/retailer_api/admin/retailer.ex
defmodule RetailerApi.Admin.Retailer do
  use Ecto.Schema
  import Ecto.Changeset

  schema "retailers" do
    # field :id, :integer
    field :GSTIN, :string
    field :Business_name, :string
    field :Contact_person, :string
    field :Contact_number, :integer
    field :Contact_address, :string
    field :Contact_emailId, :string
    field :Status, :string
    field :Outlet_limit, :integer

    # timestamps(type: :utc_datetime)
  end

  @doc false
  def changeset(retailer, attrs) do
    retailer
    |> cast(attrs, [:GSTIN, :Business_name, :Contact_person, :Contact_number, :Contact_address, :Contact_emailId, :Status, :Outlet_limit])
    |> validate_required([:GSTIN, :Business_name, :Contact_person, :Contact_number, :Contact_address, :Contact_emailId, :Status, :Outlet_limit])
  end
end
